File: DogShelter/DogShelter.Model/Posjeta.cs
namespace DogShelter.Model;

public class Posjeta
{
    public int PosjetaId { get; set; }
    public int KorisnikId { get; set; }
    public string? KorisnikIme { get; set; }
    public string? KorisnikPrezime { get; set; }
    public int? PasId { get; set; }
    public string? PasNaziv { get; set; }
    public string? PasSlikaNaslovna { get; set; }
    public DateTime DatumVrijeme { get; set; }
    public int StatusPosjeteId { get; set; }
    public string? StatusPosjeteNaziv { get; set; }
    public string? Napomena { get; set; }
    public DateTime DatumKreiranja { get; set; }
    public DateTime? DatumObrade { get; set; }
    public int? ObradioKorisnikId { get; set; }
    public string? ObradioKorisnikIme { get; set; }
    public string? ObradioKorisnikPrezime { get; set; }
    public string? RazlogOtkazivanja { get; set; }
}

File: dogshelter_mobile/lib/features/visits/domain/posjeta.dart
class Posjeta {
  Posjeta({
    required this.posjetaId,
    required this.korisnikId,
    this.korisnikIme,
    this.korisnikPrezime,
    this.pasId,
    this.pasNaziv,
    this.pasSlikaNaslovna,
    required this.datumVrijeme,
    required this.statusPosjeteId,
    this.statusPosjeteNaziv,
    this.napomena,
    required this.datumKreiranja,
    this.datumObrade,
    this.obradioKorisnikId,
    this.obradioKorisnikIme,
    this.obradioKorisnikPrezime,
    this.razlogOtkazivanja,
  });

  final int posjetaId;
  final int korisnikId;
  final String? korisnikIme;
  final String? korisnikPrezime;
  final int? pasId;
  final String? pasNaziv;
  final String? pasSlikaNaslovna;
  final DateTime datumVrijeme;
  final int statusPosjeteId;
  final String? statusPosjeteNaziv;
  final String? napomena;
  final DateTime datumKreiranja;
  final DateTime? datumObrade;
  final int? obradioKorisnikId;
  final String? obradioKorisnikIme;
  final String? obradioKorisnikPrezime;
  final String? razlogOtkazivanja;

  factory Posjeta.fromJson(Map<String, dynamic> json) {
    return Posjeta(
      posjetaId: json['posjetaId'] as int,
      korisnikId: json['korisnikId'] as int,
      korisnikIme: json['korisnikIme'] as String?,
      korisnikPrezime: json['korisnikPrezime'] as String?,
      pasId: json['pasId'] as int?,
      pasNaziv: json['pasNaziv'] as String?,
      pasSlikaNaslovna: json['pasSlikaNaslovna'] as String?,
      datumVrijeme: DateTime.parse(json['datumVrijeme'] as String),
      statusPosjeteId: json['statusPosjeteId'] as int,
      statusPosjeteNaziv: json['statusPosjeteNaziv'] as String?,
      napomena: json['napomena'] as String?,
      datumKreiranja: DateTime.parse(json['datumKreiranja'] as String),
      datumObrade: json['datumObrade'] == null ? null : DateTime.parse(json['datumObrade'] as String),
      obradioKorisnikId: json['obradioKorisnikId'] as int?,
      obradioKorisnikIme: json['obradioKorisnikIme'] as String?,
      obradioKorisnikPrezime: json['obradioKorisnikPrezime'] as String?,
      razlogOtkazivanja: json['razlogOtkazivanja'] as String?,
    );
  }
}
